### ISSUE TITLE
[Feature Request] No spin limit in Battle Café

### ISSUE THREAD

**Issue by Ignorance001** [Ephenia/Pokeclicker-Scripts#256]
Hello,

I don't know if it's possible but a script removing the 10 spins per day limitation in Battle Café would be awesome.

Since you need at least 51 spins for the 64 Pokémon available only in Battle Café to get them resistant to Pokerus, it would require more than 320 days with the current limit. A script removing this limit would save a huge amount of time.

**Comment by Ephenia** [comment]
Just dropping in as also a side note as well here.

Progression early to end game should be at max 3 months for full completion, well for healthy progression for a game. I say this because basically a year required especially for something like this is silly. It's going to take most people longer than that, since most people take breaks after some weeks/few months, or merely don't have an attention span like that.

So if anyone is taking notes for a PR that's more towards making something balanced for changes, then that would be ideal for some changes. Have something that removes the limit entirely, or base it around 3 months. Either, or, both pass my approval.

I haven't been keeping up with the game so much in recent times, so apologies. I have no idea what the Battle Café is, lol.

**Comment by K0STYURIK** [comment]
Yeah, reasonable request. When I played I made six different Alcremie "legitimate" way and the just used a console code to add the rest 57

**Comment by Seles13** [comment]
You don't need 320 days to complete alcremie resistant.
If you use farm + blue flute ( bonus 10% EV) + Power Bracer, you need just 9 alcremie to get 1 resistant.
If you add 1 to capture it, that mean you need 10 alcremie to catch/resistant.
64*10 = 640 spin to get full alcremie + cheese.
You take 13 Day to capture every alcremie to get 3 Spin to 10 => 68 Spin at this point where 4 spin need to begin EV.
After that you need 572 spin so 58 Days.
So if you don't miss spin, it require 71 days to catch/resistant Alcremie.

You need 4.2M berry and it take over easy 1 or 2 month to grow everything and you need to use you farm at least 3 hour every 2 day to put EV setup just before 00:00 to spin 2 day in a row to avoid everyday put the setup and loose 3 hour....

So even you can spin with no limit, you won't gain a lot of time. I spend h24 with best setup to get 4.2M berry and i finish with just 20/25 day ahead.

**Comment by Synogun** [comment]
need to bring this up again, my idea is instead of removing/altering limits, how about a auto spin?
should be vanilla friendly enough for those who want it and add some non-vanilla friendly possibilities!

**Comment by VariablePwn** [comment]
Hello lovely humans!
Through the magic of ChatGPT, i bring you a very basic script that injects a button to the Battle Cafe which allows you to reset the spins back to 3. I used the desktop version, and have limited js knowledge to ensure it is safe and fully function (proceed at own risk), but it may be a starting point for someone more knowledgeable.
If I wasn't supposed to post this here, I apologise in advance!

Heres the magic:

    // ==UserScript==
    // @name         Pokéclicker Battle Café Spin Reset (Final Fix)
    // @namespace    pokeclicker
    // @version      1.2
    // @description  Resets Battle Café spins
    // @author       VariablePwn + ChatGPT
    // @match        *://*/*
    // @run-at       document-idle
    // @grant        none
    // ==/UserScript==
      
      (function () {
          'use strict';
      
          // Wait for App and BattleCafeController
          const waitForGame = setInterval(() => {
              if (typeof BattleCafeController !== 'undefined' && BattleCafeController.spinsLeft) {
                  clearInterval(waitForGame);
                  injectResetButton();
              }
          }, 500);
      
          function injectResetButton() {
              // Observe DOM changes to place the button when Battle Café opens
              const observer = new MutationObserver(() => {
                  const modalTitle = document.querySelector('#battleCafeModal .modal-title');
                  if (modalTitle && !document.querySelector('#resetSpinsBtn')) {
                      const btn = document.createElement('button');
                      btn.id = 'resetSpinsBtn';
                      btn.className = 'btn btn-sm btn-danger ml-2';
                      btn.textContent = 'Reset Spins';
      
                      btn.onclick = () => {
                          // Reset spins using the true game controller
                          BattleCafeController.spinsLeft(BattleCafeController.baseDailySpins);
      
                          // Notify the user (using in-game system)
                          Notifier.notify({
                              message: 'Battle Café spins reset to 3.',
                              type: NotificationConstants.NotificationOption.success,
                              sound: NotificationConstants.NotificationSound.generic
                          });
                      };
      
                      modalTitle.appendChild(btn);
                  }
              });
      
              observer.observe(document.body, { childList: true, subtree: true });
          }
      })();
